Antidote/AbstractCallViewController.h

```diff
@@ -18,21 +16,20 @@
 + (instancetype)new NS_UNAVAILABLE;
 
 /**
- * Create an instance of the CallViewController.
- * @param chat Appropriate chat for the call.
- * @param manager The call maanger for the application.
+ * Create an instance of the AbstractCallViewController.
+ * @param nickname The nickname associated with the caller.
  */
-- (instancetype)initWithCall:(OCTCall *)call submanagerCalls:(OCTSubmanagerCalls *)manager;
+- (instancetype)initWithCallerNickname:(NSString *)nickname;
 
 /**
- * The call associated with this view controller.
+ * Abstract delegate
  */
-@property (strong, nonatomic, readonly) OCTCall *call;
+@property (weak, nonatomic) id delegate;
 
 /**
- * The call manager responsible for call handling.
+ * Name of the caller
  */
-@property (weak, nonatomic) OCTSubmanagerCalls *manager;
+@property (strong, nonatomic, readonly) NSString *nickname;
```

Comment:
Aren't you planning to change `nickname` of controller when switching call? In this case there is no need in `readonly`, as well as in `initWithCallerNickname` method. You can create controller and then set `nickname` property